### ISSUE TITLE
 Request not successful. Message: Request-URI Too Long

### ISSUE THREAD

**Issue by kimusan** [uwolfer/gerrit-intellij-plugin#196]
I am trying to get this plugin to work with our local gerrit server. In the settings it says that it connects successfully and I get a nice notification balloon with changes I need to review. Short aft er that I see this error:
Failed to get Gerrit changes.: Request not successful. Message: Request-URI Too Long. Status-Code: 414. Content: .
And I get the same right away if I press the reload button over the list. 

This is for a project with many git projects connected so that might be a reason for this but I am not sure how to verify it as I dont have any projects with less git-projects integrated.

**Comment by uwolfer** [comment]
@kimusan: Yes, this is most probably because of your many Git projects. Please try the this workaround: open Gerrit plugin settings, and enable "List all Gerrit changes (instead of changes from currently open projects only)" checkbox.

**Comment by uwolfer** [comment]
@kimusan: Please check out my workaround above and post any feedback. Thanks.

**Comment by kimusan** [comment]
Hi sorry for the delay. The workaround seems to work ok. I do however not get the diff/changes parts. It says that the project does not exist, which is not correct.

**Comment by uwolfer** [comment]
@kimusan: Thanks for your feedback. How many Gerrit Git repos do you have in your current IntelliJ project? Does diffing not work for any change of any repo? What is the exact error message?

**Comment by kimusan** [comment]
I have i.e the entire android AOSP source tree so that means probably 300-750 projects and several tags and branches on each. 
There is not really that much error shown. The window that should show the difficulty empty and as I remember it (not able to test this week) it says somewhere that there is no matching project but I can't remember the exact wording.

**Comment by uwolfer** [comment]
@kimusan: Okay, I understand now. The query for filtering that many projects will get way too long...

You could try point 2 and / or 3 of this tips: https://github.com/uwolfer/gerrit-intellij-plugin#list-of-changes-is-empty

Please let me know if that helps once you get the chance to try. Thanks.

**Comment by uwolfer** [comment]
@kimusan: Have you had time to look into this issue?

**Comment by erikhakansson** [comment]
Any updates to this?

The workaround works, but that means having to search through thousands of changes instead of the 10-20 relevant.

**Comment by erikhakansson** [comment]
I suppose a proper fix for this would be to split up requests for lots of projects into multiple requests.
